File: frontend/lib/models/product.dart

class Product {
  String id;
  String name;
  String stock;
  String price;
  String imageurl;

  Product({
    required this.id,
    required this.name,
    required this.stock,
    required this.price,
    required this.imageurl,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      stock: json['stock'],
      price: json['price'],
      imageurl: json['imageurl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'stock': stock,
      'price': price,
      'imageurl': imageurl,
    };
  }
}

File: server/src/models/product.ts
import { DataTypes } from "sequelize";
import db from "../../db/connection";

export const Product = db.define('product', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    name: {
        type: DataTypes.STRING
    },
    stock: {
        type: DataTypes.INTEGER
    },
    price: {
        type: DataTypes.FLOAT
    },
    imageurl: {
        type: DataTypes.STRING
    }
},
    {
        freezeTableName: true
    })